### PR TITLE
[Feat] Quests overlay reward claim

### PR DESCRIPTION
**Pull request by BrettCleary**
# Summary
This one got a lil big. ~600 lines. It is mostly feature flagged new features though

This PR
- Adds quest reward minting in the overlay with wagmi
- Adds wagmi to index.tsx so it can be used in the future in the main window as well
- Upgrades to `@tanstack/react-query` which is the new package name for react-query v4
- Adds support for HP account sign in through the overlay
- Refactors `QuestViewer` into `QuestDetailsWrapper` and `QuestLogWrapper` since it was getting bloated after adding the minting code

# Note
The UX is not perfect and that is not the goal of this PR. There will be follow up PR's for 
- getting claimable/claimed status on initial page load
- overlay hiding after confirming switch chain/approve txn
- adding support for more chains in wagmi

# Testing
Test with the overlay in Kosium

### CODE REVIEW

**Review by flavioislima — APPROVED** [review]
Good stuff

File: src/backend/api/auth.ts [hunk 1]
```diff
@@ -12,3 +12,15 @@ export const logOut = async () => {
 
 export const signInWithProvider = (provider: string) =>
   ipcRenderer.send('openExternalUrl', `${DEV_PORTAL_URL}/oauth/${provider}`)
+
+export const authConnected = () => ipcRenderer.send('authConnected')
+export const authDisconnected = () => ipcRenderer.send('authDisconnected')
```
Comment (eliobricenov) [inline src/backend/api/auth.ts:17]: What do you think about renaming these to `notifyWindowsAuthConnected` and `notifyWindowsauthDisconnected`?

This is what we're doing right?

**Review by eliobricenov — APPROVED** [review]
lgtm, great work

I left a couple suggestions that can come up in a follow up PR

**Review by red-game-dev — APPROVED** [review]
LGTM - Checked only the code

**Comment by nyghtstalker** [comment]
1. All font is larger

**Comment by nyghtstalker** [comment]
1. All font is larger

image
image

2. Attempting to claim with the wallet locked has no effect. I would recommend some way of alerting the user to unlock their wallet.

------------------

Other than the above I was able to claim quests and sign into my email from the overlay.

image

**Review by nyghtstalker — APPROVED** [review]

File: src/backend/api/auth.ts (hunk 1 above)
Comment (BrettCleary) [reply src/backend/api/auth.ts:17]: it is notifying the backend only. the backend decides whether to notify other windows or not

**Comment by BrettCleary** [comment]
> lgtm, great work
> 
> I left a couple suggestions that can come up in a follow up PR

I will definitely handle these on a follow-up PR 🙏

**Pull request merged**